packages/sakuli-commons/src/logger/stringifier/default-stringifier.function.ts

```diff
@@ -18,7 +18,7 @@ const stringifyData = (data: string | number | object) => {
 };
 
 export const defaultStringifier: LogEventStringifier = (e: LogEvent) => {
-  return `[${e.time}] ${LogLevel[e.level]}: ${e.message}${EOL}${e.data
-    .map(stringifyData)
-    .join(EOL)}`;
+  return `[${e.time}] ${LogLevel[e.level]}: ${e.message}${EOL}${
+    e.data.length === 0 ? "" : `${e.data.map(stringifyData).join(EOL)}${EOL}`
+  }`;
```

Comment:
If we change ordering for the ternary operator, the zero-length check could be omitted.
Since the result of joining an empty array is an empty string, the conditional could be re-written to e.g.

```typescript
    return `[${e.time}] ${LogLevel[e.level]}: ${e.message}${EOL}${e.data
        .map(stringifyData)
        .join(EOL)
        .concat(`${e.data.length ? EOL : ''}`)}`;
```